modules/sharedaddy/sharing-sources.php

```diff
@@ -1701,7 +1701,17 @@ public function get_display( $post ) {
 	public function process_request( $post, array $post_data ) {
 		// Record stats
 		parent::process_request( $post, $post_data );
-		$url = 'https://api.whatsapp.com/send?text=' . rawurlencode( $this->get_share_title( $post->ID ) . ' ' . $this->get_share_url( $post->ID ) );
+
+		$ua = strtolower( $_SERVER['HTTP_USER_AGENT'] );
```

Comment:
I think it would make sense to use `Jetpack_User_Agent_Info` here, and create a new function in that class to detect Firefox for Desktop, just like we do with `is_firefox_mobile()` for example.
https://github.com/Automattic/jetpack/blob/8d9107fbc1f7d04446a24444c314b8e56e04f7df/class.jetpack-user-agent.php#L685

Reply:
Good point! The code is cleaner and more explicit now.